src/server/utilities.ts

```diff
@@ -90,6 +90,21 @@ namespace ts.server {
         };
     }
 
+    export function getLatestDirectoryChangeTime(paths: string[] | undefined, host: System) {
+        if (!host.getModifiedTime || !host.directoryExists || !paths) {
+            return 0;
+        }
+
+        return Math.max.apply(Math, paths.map(path => {
+            if (host.directoryExists(path)) {
+                return host.getModifiedTime(path);

```

Comment:
This returns a Date, but this function should return a number (Math.max probably converts it implicitly here).